
<!-- saved from url=(0124)http://git.gnome.org/browse/gtk-engines/patch/engines/clearlooks/src/animation.c?id=a484a86cef26420e9278a550d246ff1068a9e88e -->
<html><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><style type="text/css"></style></head><body><pre style="word-wrap: break-word; white-space: pre-wrap;">From a484a86cef26420e9278a550d246ff1068a9e88e Mon Sep 17 00:00:00 2001
From: Benjamin Berg &lt;benjamin@sipsolutions.net&gt;
Date: Fri, 02 Jul 2010 16:56:24 +0000
Subject: Fix animation. (bug #622982, #612486)

---
(limited to 'engines/clearlooks/src/animation.c')

diff --git a/engines/clearlooks/src/animation.c b/engines/clearlooks/src/animation.c
index 1f7a41f..d37b85d 100644
--- a/engines/clearlooks/src/animation.c
+++ b/engines/clearlooks/src/animation.c
@@ -166,7 +166,7 @@ update_animation_info (gpointer key, gpointer value, gpointer user_data)
 	g_assert ((widget != NULL) &amp;&amp; (animation_info != NULL));
 	
 	/* remove the widget from the hash table if it is not drawable */
-	if (!GTK_WIDGET_DRAWABLE (widget))
+	if (!gtk_widget_is_drawable (widget))
 	{
 		return TRUE;
 	}
@@ -267,6 +267,20 @@ find_signal_info (gconstpointer signal_info, gconstpointer widget)
 
 /* external interface */
 
+#define CL_IS_PROGRESS_BAR(widget) GE_IS_PROGRESS_BAR(widget) &amp;&amp; widget-&gt;allocation.x != -1 &amp;&amp; widget-&gt;allocation.y != -1
+gboolean
+clearlooks_animation_is_progressbar (GtkWidget *widget)
+{
+	GtkAllocation allocation;
+	if (!GE_IS_PROGRESS_BAR (widget))
+		return FALSE;
+	gtk_widget_get_allocation (widget, &amp;allocation);
+	if ((allocation.x &lt; 0) || (allocation.y &lt; 0))
+		return FALSE;
+	return TRUE;
+}
+
+
 /* adds a progress bar */
 void
 clearlooks_animation_progressbar_add (GtkWidget *progressbar)
@@ -331,8 +345,6 @@ clearlooks_animation_cleanup ()
 	stop_timer ();
 }
 #else /* !HAVE_WORKING_ANIMATION */
-/* Warn here so the message is only displayed once. */
-#warning Disabling animation support as it currently needs deprecated symbols and GTK_DISABLE_DEPRECATED is enabled.
 
 static void clearlooks_animation_dummy_function_so_wall_shuts_up_when_animations_is_disabled()
 {
--
cgit v0.9.0.2
</pre></body></html>